src/parser/jobs/dnc/modules/DirtyDancing.tsx

```diff
@@ -197,9 +197,12 @@ export default class DirtyDancing extends Module {
 		dance.end = finisher.timestamp
 
 		// Count dance as dirty if we didn't get the expected finisher, and the fight wouldn't have ended or been in an invuln window before we could have
-		if (finisher.ability.guid !== dance.expectedFinishId && !(this.parser.fight.end_time < dance.expectedEndTime || this.invuln.isInvulnerable('all', dance.expectedEndTime))) {
-			dance.dirty = true
+		if (finisher.ability.guid !== dance.expectedFinishId && dance.expectedEndTime <= this.parser.eventTimeOffset + this.parser.pull.duration) {
+			this.addTimestampHook(dance.expectedEndTime, ({timestamp}) => {
+				dance.dirty = !this.invuln.isInvulnerable('all', timestamp)
+			})
```

Comment:
dance dirty status isn't read until end of fight.
dance overshooting end of fight is fine, so narrowing that case out and just hooking cases where it wasn't to check invuln

Reply:
Yep, this change looks good to me 👍